pysmt/cmd/installers.py

```diff
@@ -0,0 +1,630 @@
+        u = urllib2.urlopen(url)
+        f = open(file_name, 'wb')
+        meta = u.info()
+        if meta.get("Content-Length") and len(meta.get("Content-Length")) > 0:
+            file_size = int(meta.get("Content-Length"))
+            print("Downloading: %s Bytes: %s" % (file_name, file_size))
+
+        block_sz = 8192
+        count = 0
+        while True:
+            buff = u.read(block_sz)
+            if not buff:
+                break
+
+            f.write(buff)
+            if meta.get("Content-Length") and len(meta.get("Content-Length")) > 0 \
+               and sys.stdout.isatty():
+                count += len(buff)
+                perc = (float(count) / float(file_size)) * 100.0
+                str_perc = "%.1f%%" % perc
+                sys.stdout.write('\r')
+                sys.stdout.write(str_perc)
+                sys.stdout.write(" " * (10 - len(str_perc)))
+
+        print("")
+        f.close()
+        return True
+
+
+    @staticmethod
+    def run_python(script, directory=None, env_variables=None):
+        """Executes a python script"""
+        interpreter = 'python'
+        if sys.executable:
+            interpreter = sys.executable
+
+        cmd = '{interpreter} {script}'.format(interpreter=interpreter,
+                                              script=script)
+        SolverInstaller.run(cmd, directory=directory, env_variables = env_variables)
+
+    @staticmethod
+    def run(program, directory=None, env_variables=None):
+        """Executes an arbitrary program"""
+        if directory is not None:
+            cmd = 'cd {directory}; {program}'
+        else:
+            cmd = '{program}'
+
+        if env_variables is not None:
+            for k,v in env_variables.iteritems():
+                cmd = "export %s='%s'; %s" % (k, v, cmd)
+
+        os.system(cmd.format(directory=directory,
+                             program=program))
+
+    @staticmethod
+    def clean_dir(path):
+        """Empties a (possibly non-existent) directory"""
+        if os.path.exists(path):
+            shutil.rmtree(path)
+        os.mkdir(path)
+
+    @staticmethod
+    def mv(source_file, dest):
+        """Similarly to the UNIX mv command, moves / renames source_file in
+        dest (if dest is a file name) otherwise moves source_file in
+        the directory dest
+        """
+        if os.path.isdir(dest):
+            fname = os.path.basename(source_file)
+            dest_file = os.path.join(dest, fname)
+        else:
+            dest_file = dest
+
+        if os.path.isfile(dest_file):
+            os.unlink(dest_file)
+        os.rename(source_file, dest_file)
+
+
+class MSatInstaller(SolverInstaller):
+
+    SOLVER = "msat"
+
+    def __init__(self, install_dir, bindings_dir, solver_version, os_name=None,
+                 architecture=None, python_version=None, mirror_link=None):
+        SolverInstaller.__init__(self, install_dir, bindings_dir, solver_version,
```

Comment:
Use keyword arguments. There are too many parameters, and we will eventually mix-up some of them.